openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -596,8 +595,11 @@ def password_change_request_handler(request):
 
     """
     user = request.user
-    # Prefer logged-in user's email
-    email = user.email if user.is_authenticated else request.POST.get('email')
+    if user.is_staff and user.is_superuser and request.POST.get('email_from_support_tools'):
```

Comment:
check if there is an existing method within platform codebase that does the staff and superuser check.

Reply:
Unfortunately, I was unable to find one. I think I might've seen one but don't remember which method was it.